week4-18-02-19.md

```diff
@@ -7,10 +7,16 @@
 HELP NEEDED:
 only leaving this one in without a description right now because it's important to know. A description should be written
 --->
+In previous betas of Vapor 3, the service framework supported an `isSingleton` property for services which controlled their behavior when requested from different containers and/or from different requesting interfaces (the `for:` parameter of `Container.make()`). Singletons would in theory have only the single instance across all invocations. However, this only held true for services whose underlying implementations were `class`es; `struct`-based services were never singletons even when the flag was specified. In addition, the flag offered a manual control for a behavior already baked into Swift via the difference between value and reference types. Therefore, the singleton flag on service types was removed entirely. Now, to get the "singleton" behavior, use a `class` type to implement your service. `struct`-based services now receive the previously default "instanced" behavior (a new instance of the service per request).
+
+TL;DR: The redundant singleton flag on services was removed, and the behavior when requesting a given service more than once now corresponds to Swift's standard value vs. reference semantics.
 
 [vapor/service#14](https://github.com/vapor/service/pull/14)
 
-### Both `struct` and `class` SQLite Models No Longer Require `init` Methods To Be Defined
+### SQLite Models No Longer Require `init` Methods
+
+This is true for both `struct`- and `class`-based models.
```

Comment:
Uh...what? How is that happening for classes.

Reply:
oh uhhhh. I completely misread the PR. What actually changed?
https://github.com/vapor/fluent/pull/380